arbitrum-docs/stylus/reference/project-structure.mdx

```diff
@@ -0,0 +1,209 @@
+
+**Rust-style Schema**
+
+```rust
+use stylus_sdk::{prelude::*, storage::{StorageU256, StorageAddress}};
+
+#[storage]
+#[entrypoint]
+pub struct MyContract {
+    owner: StorageAddress,
+    version: StorageU256,
+}
+```
+
+**Solidity-style Schema**
+
+```rust
+use stylus_sdk::{prelude::*};
+
+sol_storage! {
+    #[entrypoint]
+    pub struct MyContract {
+        address owner;
+        version: uint256,
+    }
+}
+```
+
+To read from state or write to it, getters and setters are used:
+
+```rust
+let new_count = self.count.get() + U256::from(1);
+self.count.set(new_count);
+```
+
+## Functions
+
+Contract functions are defined by providing an `impl` block for your contract's `#[entrypoint]` struct and annotating that block with `#[public]` to make the functions part of the contract's public API. The first parameter of each function is `&self`, which references the struct annotated with `#[entrypoint]`, it's used for reading state variables. By default, methods are view-only and cannot mutate state. To make a function mutable and able to alter state, `&mut self` must be used. Internal methods can be defined on a separate impl block for the struct that is not annotated with `#[public]`. Internal methods can access state.
+
+```rust
+// Defines the public, external methods for your contract
+// This impl block must be for the #[entrypoint] struct defined prior
+#[public]
+impl Counter {
+    // By annotating first arg with &self, this indicates a view function
+    pub fn get(&self) -> U256 {
+        self.count.get()
+    }
+
+    // By annotating with &mut self, this is a mutable public function
+	pub fn set_count(&mut self, count: U256) {
+        self.count.set(count);
+    }
+}
+
+// Internal methods (NOT part of public API)
+impl Counter {
+    fn add(a: U256, b: U256) -> U256 {
+        a + b
+    }
+}
+```
+
+## Modules
+
+Modules are a way to organize code into logical units. While your contract must have a `lib.rs` which defines your entrypoint struct, you can also define utility functions, structs, enums, etc., in modules and import them to use in your contract's methods.
+
+For example, with this file structure:
+
+```
+- src
+    - lib.rs
+    - main.rs
+- utils
+    - mod.rs
+- Cargo.toml
+- rust-toolchain.toml
+```
+
+In `lib.rs`:
+
+```rust
+// import module
+mod utils;
+
+// ..other code
+const score = utils::check_score();
+```
+
+See [Defining modules](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html) in the Rust book for more info on modules and how to use them.
+
+## Importing packages
+
+Rust has a robust package manager for managing dependencies and importing third-party libraries to use in your smart contracts. These packages (called crates in Rust) are located at [crates.io](https://crates.io). To make use of a dependency in your code, you'll need to complete these steps:
```

Comment:
Worth mentioning that some crates will not work in Stylus contracts because they compile to WASM and are used in a blockchain context. "For instance, the `rand` crate, as there is no onchain randomness available to contracts. Also, functions that involve networking, filesystem access, etc." Maybe we can also tell people that to figure out if a dependency is 100% compatible, they can check if it requires the Rust standard library. No_std crates are always compatible with Stylus contracts